src/extension/fake_id0/helper_functions.c

```diff
@@ -317,19 +340,49 @@ int read_meta_file(char path[PATH_MAX], mode_t *mode, uid_t *owner, gid_t *group
 {
 	FILE *fp;
 	int lcl_mode;
-	fp = fopen(path, "r");
-	if(!fp) {
-		/* If the metafile doesn't exist, allow overly permissive behavior. */
-		*owner = config->euid;
-		*group = config->egid;
-		*mode = otod(755);
-		return 0;
+	int status;
```

Comment:
I think you might have an easier time removing metafile functionality during a later refactor if you extracted the leveldb read into its own routine, and did metafile reading fallback if that routine fails. At the very least, I think it would be useful to have clearer semantics concerning what read_meta_file does. It's called in a lot of places, but "reading a meta file" isn't really the intent of this function anymore.